src/hotspot/share/code/aotCodeCache.cpp

```diff
@@ -84,10 +83,30 @@ static void report_store_failure() {
     vm_abort(false);
   }
   log_info(aot, codecache, exit)("Unable to create AOT Code Cache.");
-  AOTAdapterCaching = false;
-  AOTStubCaching = false;
+  AOTCodeCache::disable_caching();
 }
 
+// The sequence of AOT code caching flags and parametters settings.
+//
+// 1. The initial AOT code caching flags setting is done
+// during call to CDSConfig::check_vm_args_consistency().
+//
+// 2. The earliest AOT code state check done in compilationPolicy_init()
+// where we set number of compiler threads for AOT assembly phase.
+//
+// 3. We determine presence of AOT code in AOT Cache in
+// MetaspaceShared::open_static_archive() which is calles
+// after compilationPolicy_init() but before codeCache_init().
+//
+// 4. AOTCodeCache::initialize() is called during universe_init()
```

Comment:
@vnkozlov I wonder if we can move `AOTCodeCache::initialize()` after universe_init() and merge `AOTCodeCache::init2()` into `AOTCodeCache::initialize()`. Is there any reason for splitting initialization between `initialize()` and  `init2`?

Reply:
> We need to load/map AOT code region before we close file

right, I missed that part.